### PR TITLE
feat: Split compare / check to two different baseline commands

### PR DESCRIPTION
**Pull request by hassila**
## Description

Split `compare` and threshold violation checking into a new command `check`.

`compare` will show delta as previously, while `check` output more detailed and parsable deviation information.

## How Has This Been Tested?

Manual testing.

## Minimal checklist:

- [x] I have performed a self-review of my own code 
- [x] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [x] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: Plugins/Benchmark/BenchmarkPlugin.swift [hunk 1]
```diff
@@ -241,36 +235,50 @@ import PackagePlugin
...
-                case "read": // to allow for a baseline named 'update'
-                    positionalArguments.removeFirst()
-                case "list":
-                    positionalArguments.removeFirst()
-                    args.append(contentsOf: ["--list-baselines"])
-                default:
-                    break
+            switch baselineOperation {
+            case .update:
+                args.append(contentsOf: ["--update"])
```
Comment (dimlio) [inline Plugins/Benchmark/BenchmarkPlugin.swift:252]: Why not simply `args.append("--update")` ?

File: Plugins/BenchmarkTool/BenchmarkTool+Baselines.swift [hunk 2]
```diff
@@ -182,10 +182,7 @@ extension BenchmarkTool {
                 var subDirectory = storagePath
                 subDirectory.append(file.lastComponent!)
                 if let directoryName = file.lastComponent {
```
Comment (dimlio) [inline Plugins/BenchmarkTool/BenchmarkTool+Baselines.swift:184]: Hmm, the line above `file.lastComponent` was force-unwrapped so this check should always succeed, or?

File: Plugins/BenchmarkTool/BenchmarkTool+Operations.swift [hunk 3]
```diff
@@ -96,16 +96,64 @@ extension BenchmarkTool {
...
 
                 prettyPrintDelta(currentBaseline: currentBaseline, baseline: comparisonBaseline)
 
-                if currentBaseline.betterResultsOrEqual(than: comparisonBaseline, printOutput: true) {
+                return
+            }
+
+            if let checkBaseline {
+                guard benchmarkBaselines.count > 0 else {
+                    print("Only had \(benchmarkBaselines.count) baselines available, can't check.")
```
Comment (dimlio) [inline Plugins/BenchmarkTool/BenchmarkTool+Operations.swift:107]: But the only possible value is 0, maybe just say `"No baselines available, cat't check"`?

File: Plugins/BenchmarkTool/BenchmarkTool.swift [hunk 4]
```diff
@@ -71,6 +71,9 @@ struct BenchmarkTool: AsyncParsableCommand {
     @Option(name: .long, help: "The baseline to compare with")
     var compare: String?
 
+    @Option(name: .long, help: "The baseline to check thresholds with")
+    var check: String?
+
     @Flag(name: .long, help: "True if we should supress output")
```
Comment (dimlio) [inline Plugins/BenchmarkTool/BenchmarkTool.swift:77]:
```suggestion
    @Flag(name: .long, help: "True if we should suppress output")
```

**Review by dimlio — APPROVED** [review]

File: Plugins/BenchmarkTool/BenchmarkTool+Baselines.swift (hunk 2 above)
Comment (hassila) [reply Plugins/BenchmarkTool/BenchmarkTool+Baselines.swift:183]: Yes, refactored both of these force unwraps properly now with proper `let` extraction instead, thanks.

**Pull request merged**